### PR TITLE
Fix sonarqube issues

### PR DESCRIPTION
**Pull request by Gr33ndev**
Part of Gamify-IT/issues#498

https://sonar.sqa-kube-cluster4.cloudns.nz/dashboard?id=towercrush-backend

**Please test is everything is still working as intendet**

### CODE REVIEW

File: src/main/java/de/unistuttgart/towercrushbackend/controller/components/WebsocketListener.java [hunk 1]
```diff
@@ -122,7 +123,7 @@ private void handleDeveloperJoined() {
                     try {
                         Thread.sleep(1000);
                     } catch (final Exception e) {
-                        throw new RuntimeException(e);
+                        throw new InterruptedException();
                     }
```
Comment (delvh) [inline src/main/java/de/unistuttgart/towercrushbackend/controller/components/WebsocketListener.java]: Erm…
That won't work for two reasons:
1. `Thread.sleep` already throws `InterruptedException`
2. That is a checked exception that you cannot throw here.

File: src/main/java/de/unistuttgart/towercrushbackend/data/Configuration.java [hunk 2]
```diff
@@ -38,7 +38,7 @@ public void removeQuestion(final Question question) {
     }
 
     @Override
-    public Configuration clone() {
+    public Configuration clone() { //NOSONAR
```
Comment (delvh) [inline src/main/java/de/unistuttgart/towercrushbackend/data/Configuration.java]: Not needed here?

File: src/main/java/de/unistuttgart/towercrushbackend/data/Question.java [hunk 3]
```diff
@@ -38,7 +38,7 @@ public Question(final String text, final String rightAnswer, final Set<String> w
     }
 
     @Override
-    public Question clone() {
+    public Question clone() { //NOSONAR
```
Comment (delvh) [inline src/main/java/de/unistuttgart/towercrushbackend/data/Question.java]: Still not needed.

File: src/main/java/de/unistuttgart/towercrushbackend/data/Configuration.java (hunk 2 above)
Comment (delvh) [reply src/main/java/de/unistuttgart/towercrushbackend/data/Configuration.java]:
```suggestion
    public Configuration clone() {
```

File: src/main/java/de/unistuttgart/towercrushbackend/data/Question.java (hunk 3 above)
Comment (delvh) [reply src/main/java/de/unistuttgart/towercrushbackend/data/Question.java]:
```suggestion
    public Question clone() {
```

File: src/main/java/de/unistuttgart/towercrushbackend/service/websockets/GameService.java [hunk 4]
```diff
@@ -83,11 +83,10 @@ public void createGame(final String lobby, final UUID configurationId) {
...
             tempRounds,
             configurationId,
-            (long) tempRounds.size() * timePerQuestion
+            (long) tempRounds.size() * TIME_PER_QUESTION
+
         );
-        if (!games.containsKey(lobby)) {
-            games.put(lobby, game);
-        }
+        games.computeIfAbsent(lobby, k -> games.put(lobby, game));
```
Comment (delvh) [inline src/main/java/de/unistuttgart/towercrushbackend/service/websockets/GameService.java]:
```suggestion
        games.computeIfAbsent(lobby, k -> game);
```

File: src/main/java/de/unistuttgart/towercrushbackend/service/websockets/GameService.java [hunk 5]
```diff
@@ -420,10 +419,10 @@ private void saveAndDeleteGame(final Game game) {
      *
      * @param sleepTime sleep time
      */
-    private void sleep(final int sleepTime) {
+    private void sleep(final int sleepTime){
         try {
             Thread.sleep(sleepTime);
-        } catch (final InterruptedException e) {
+        } catch (final InterruptedException e) { //NOSONAR
```
Comment (delvh) [inline src/main/java/de/unistuttgart/towercrushbackend/service/websockets/GameService.java]:
```suggestion
        } catch (final InterruptedException e) {
```
Shouldn't be needed here.

File: src/main/java/de/unistuttgart/towercrushbackend/service/websockets/GameService.java [hunk 6]
```diff
@@ -420,10 +419,10 @@ private void saveAndDeleteGame(final Game game) {
...
             log.error("could not sleep  {} seconds", sleepTime, e);
```
Comment (delvh) [inline src/main/java/de/unistuttgart/towercrushbackend/service/websockets/GameService.java]:
```suggestion
            log.error("could not sleep {} seconds. Error: {}", sleepTime, e);
```

File: src/main/java/de/unistuttgart/towercrushbackend/controller/ConfigController.java [hunk 7]
```diff
@@ -25,6 +25,7 @@
 @Slf4j
 public class ConfigController {
 
+    public static final List<String> LECTURER = List.of("lecturer");
```
Comment (delvh) [inline src/main/java/de/unistuttgart/towercrushbackend/controller/ConfigController.java]:
```suggestion
    private static final List<String> LECTURER = List.of("lecturer");
```

File: src/main/java/de/unistuttgart/towercrushbackend/controller/components/WebsocketListener.java [hunk 8]
```diff
@@ -28,6 +28,7 @@
 @Slf4j
 public class WebsocketListener {
 
+    public static final String SHA_USER_IS_NOT_SET = "sha user is not set (null)";
```
Comment (delvh) [inline src/main/java/de/unistuttgart/towercrushbackend/controller/components/WebsocketListener.java]:
```suggestion
    private static final String SHA_USER_IS_NOT_SET = "sha user is not set (null)";
```

**Review by delvh — CHANGES_REQUESTED** [review]

File: src/main/java/de/unistuttgart/towercrushbackend/data/Configuration.java (hunk 2 above)
Comment (Gr33ndev) [reply src/main/java/de/unistuttgart/towercrushbackend/data/Configuration.java]: Was neede because java:S2975, but deactivated it..

**Review by LeviUniAcc — APPROVED** [review]
looks good

**Pull request merged**